### ISSUE TITLE
Windows portable version F10 results freeze

### ISSUE THREAD

**Issue by mathtester** [pawelsalawa/sqlitestudio#5161]
### Details
Using Windows portable version 3.4.9 or 3.4.8, I cannot "Open configuration dialogue". Actually, doing so would result the SQLiteStudio freeze. I have to use Windows task manager to kill it.

### Steps to reproduce
Unzip Windows portable version. Open the program. "Tools" --> "Open configuration dialogue". Or equivalently, press F10. Then SQLiteStudio freezes. 

I feel like the dialogue is opened somewhere so the main window is blocked. But I don't know where the dialogue window is. I also tried to position the initial main SQLiteStudio window at different places. Still could not see where the dialogue is.

### Operating system
Windows 11. Never tried other operation system.

### SQLiteStudio version
Portable package type. Tried 3.4.8 and 3.4.9. The behaviors are the same.

**Comment by pawelsalawa** [comment]
Please start your application from command line using options:
```
SQLiteStudio.exe -d --debug-file=C:/path/to/existing/directory/log_file_to_create.txt
```

Then repeat your issue, kill the app when it freezes and then check the log file if there are any problems reported there.

**Comment by pawelsalawa** [comment]
3.4.9 has a bug, which sometimes placed newly open dialogs at position `0`x`0` of your desktop. If you have another monitor connected (but turned off), it might be there.

Today I'm going to release 3.4.10 which has this issue fixed, but if you already had a dialog placed at 0x0, then 3.4.10 will "restore" this 0x0 placement. The solution in that case would be to rename your old [configuration file](https://github.com/pawelsalawa/sqlitestudio/wiki/Configuration_directory_location) and start the app with empty config.

**Comment by pawelsalawa** [comment]
...and one more test you can perform. If the config window really opens up, but is just invisible, then try this:
- run the app
- press F10 to open config and don't click anywhere, so that the config dialog maintains keyboard focus
- press Esc to close the config dialog
- verify if main window resumed to normal operation

If the main window works again, then indeed the dialog opens somewhere, but you cannot see it and I would start with checking where is rest part of your desktop.

**Comment by mathtester** [comment]
I went over the above process (using 3.4.9) and indeed ESC could resume the app from freeze. So your theory is correct.

I also experienced the same situation like #5173 .  And I followed the above procedure. ESC un-freeze the app. (But the problem remains: I cannot edit a table column)

I will try version 3.4.10 to see if that solves the problem.

**Comment by pawelsalawa** [comment]
After switching to 3.4.10 - if the problem still exists, please add [your configuration file](https://github.com/pawelsalawa/sqlitestudio/wiki/Configuration_directory_location) as another database (as it actually IS just sqlite3 database file), open it and execute following SQL on it:
```
DELETE FROM settings where [group] = 'DialogDimensions';
```

The problem should be solved. Note, that this may not work while you're still working on 3.4.9 or older.

**Comment by mathtester** [comment]
I deleted my 3.4.9 folder (like C:\Users\sunhe\Documents\Data\SQLiteStudio) I downloaded portable 3.4.10. Unzip and saved into the above location. The problem is not solved.

I will try the "configuration file" trick. 

Before that, I would comment that I don't need to "press F10 to open config and don't click anywhere, so that the config dialog maintains keyboard focus". As long as the app is frozen, the focus is on the phantom window, I guess. So no matter what I do (clicking mouse, typing, etc), the only way is pressing ESC, or kill it by task manager. So operationally, testing becomes a lot easier.

**Comment by pawelsalawa** [comment]
Okay, try with config fix mentioned above and let me know if that worked.

**Comment by mathtester** [comment]
I follow the instruction about configuration file https://github.com/pawelsalawa/sqlitestudio/wiki/Configuration_directory_location
But I would not find the config folder sqlitestudio-cfg. Where is my config? I am using Windows portable installation. As I mentioned earlier, when I installed 3.4.10, I deleted the whole folder of 3.4.9 unziped. But magically, when I opened 3.4.10, my old databases are shown. This means that config folder is not located "in the same directory where SQLiteStudio executable is located" as mentioned in the instruction.

**Comment by mathtester** [comment]
OK. Got the config file. It turns out in my case, it is located here:
C:\Users\sunhe\AppData\Local\SalSoft\SQLiteStudio

**Comment by mathtester** [comment]
After I removed "settings3", I set up the databases from the scratch. Everything is good now. I can do F10, I can also edit individual columns. So it appears the bug has been fixed.

However, I remember I saw those edit screens before. So it means at some time the F10 and edit table structure stopped working. So I would continue using and see if I got into the problems later.

If the developer can give us a hint what went wrong in the code, I can do a more focused testing.

**Comment by pawelsalawa** [comment]
The bug appeared first in 3.4.5.

**Comment by pawelsalawa** [comment]
Conclusion:

Due to bug with dialog positioning that appeared by the way of fixing #4740, which was later fixed in 3.4.10, the problem remains that bad position 0x0 can still persist in broken configs. Need to flush these broken entries by deleting:
```
DELETE FROM settings WHERE [group] = 'DialogDimensions';
```
...to be done by the application automatically once during version upgrade.

Assigned this to 3.4.11 as an ultimate fix.

**Comment by mathtester** [comment]
Earlier I was puzzled why there was no such a problem in the initial uses. After used for a while, the problem appeared. Here is my understanding based on the above: 

During the initial period of use, the problematic config does not appear in the "settings3" file. So everything is fine. If somehow the broken config gets into the settings3 file, we'll see the problem.

Is this the reason for frozen app after a period of good use?

**Comment by mathtester** [comment]
Since #4740 is about resizing windows, I tested by constantly resizing all kinds of windows when I import data, config tables, etc. And periodically F10 and edit columns. I did several rounds by open/close the app. I did not experience any issues. It looks like the issue has been fixed.

Since 3.4.11 is marked as "ultimate" fix, I will wait until then to close this. Or maybe the developer will close it.
